### PR TITLE
update pause container filtering

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

- Adds broader filtering on GCR (to accommodate `eu.gcr.io/k8s-artifacts-prod/pause:3.3`) and other pause image names 
- Reorganizes pause image regexes to share a loose structure

### Motivation

- Necessary filtering of a pod image
- Make the filter constants easier to read and expand

Although, I'm not entirely sure the second motivating factor was achieved - open to other suggestions

### Describe your test plan

Make sure agents running in different providers do not report pause container tags

### CODE REVIEW

File: pkg/util/containers/filter.go [hunk 1]
```diff
@@ -14,26 +14,39 @@ import (
...
+	pauseContainerGCR = `image:(.*)gcr\.io(.*)/pause(.*)`
+
 	// - k8s-gcrio.azureedge.net/pause-amd64
 	// - gcrio.azureedge.net/google_containers/pause-amd64
-	pauseContainerAzure   = `image:(.*)azureedge\.net(/google_containers/|/)pause(.*)`
+	pauseContainerAzure = `image:(.*)azureedge\.net(.*)/pause(.*)`
+
+	// amazonaws.com/eks/pause-windows:latest
+	// eks/pause-amd64
+	pauseContainerEKS = `image:(amazonaws\.com/)?eks/pause-(amd64|windows)`
```
Comment (L3n41c) [inline pkg/util/containers/filter.go:39]: We should try to spawn an [EC2 A1 instance](https://aws.amazon.com/ec2/instance-types/a1) inside an Amazon EKS cluster, to check if we would get an `eks/pause-arm64` image ?

```suggestion
	pauseContainerEKS = `image:(amazonaws\.com/)?eks/pause-(amd64|arm64|windows)`
```

**Review by L3n41c — APPROVED** [review]
👍🏻 
The only two remarks that I would have are that we might want to check for `arm64` everywhere we check for `amd64` and that the parenthesis around `.*` are useless.

File: pkg/util/containers/filter.go (hunk 1 above)
Comment (celenechang) [reply pkg/util/containers/filter.go:39]: good point - found evidence that it exists here https://github.com/aws/containers-roadmap/issues/264

**Comment by celenechang** [comment]
thanks! i decided to keep the parentheses just in places that it helped with readability (to me anyway)

**Pull request merged**